pkg/vsphere/datastore/datastore.go

```diff
@@ -89,12 +88,13 @@ func NewHelperFromURL(ctx context.Context, s *session.Session, u *url.URL) (*Hel
 	}
 
 	d := &Helper{
-		ds:      vsDs,
-		s:       s,
-		fm:      fm,
-		RootURL: u.Path,
+		ds: vsDs,
+		s:  s,
+		fm: fm,
 	}
 
+	d.RootURL.FromString(u.Path)
+
 	return d, nil
 }
 
```

Comment:
add a `New HelperFromDatastorePath(context.Context, *session.Session, object.DatastorePath)`  so that we've got an easy transition to using object.DatastorePath directly in the config instead of URL.
The only reason we were using URL at all was for the structured path data and url.Parse - given we now have a govmomi structured path type and the url parsing is broken for spaces and unicode in the "hostname" portion it adds no value.